File: c++/hardware.cpp
class Hardware : public Product{

    //atribut
    private:
        string brand;
        string model;

    public:
        //constructor
        Hardware(){
            this->brand = "";
            this->model = "";
        }

        //setter dan getter
        void setBrand(string brand){
            this->brand = brand;
        }

        string getBrand(){
            return this->brand;
        }

        void setModel(string model){
            this->model = model;
        }

        string getModel(){
            return this->model;
        }

        //destructor
        ~Hardware(){
        }
};

File: java/Hardware.java
class Hardware extends Product{

    //atribut
    private String brand;
    private String model;

    Hardware(){
    }

    //setter dan getter
    public void setBrand(String brand){
        this.brand = brand;
    }

    public String getBrand(){
        return this.brand;
    }

    public void setModel(String model){
        this.model = model;
    }

    public String getModel(){
        return this.model;
    }
}